### PR TITLE
Support multi-repo builds

### PR DESCRIPTION
**Pull request by jmsanders**
This restructures the logic we need to conditionally skip parts of our build into classes that allow us to specify multiple git repositories. This should allow us to walk cross-repo Python dependencies.

### CODE REVIEW

File: .buildkite/dagster-buildkite/dagster_buildkite/git.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @classmethod
+    def load_repository(cls, git_repository_directory: Path) -> None:
+        # Only do the expensive git diffing once
+        if git_repository_directory in cls._repositories:
+            return None
+
+        original_directory = os.getcwd()
+        os.chdir(git_repository_directory)
+
+        # TODO: Make it possible to compare to BUILDKITE_PULL_REQUEST_BASE_BRANCH
```
Comment (jmsanders) [inline .buildkite/dagster-buildkite/dagster_buildkite/git.py:25]: This also needs to be responsive to `BUILDKITE_COMMIT` and `BUILDKITE_BRANCH` - maybe a few other things too...

File: .buildkite/dagster-buildkite/dagster_buildkite/repository.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        # Memoize
+        if git_repository_directory in cls.all:
+            return None
+
+        logging.info(f"Parsing repo {git_repository_directory.absolute()}")
+
+        ChangedFiles.load_repository(git_repository_directory)
+        PythonPackages.load_repository(git_repository_directory)
+
+        cls.all.add(git_repository_directory)
```
Comment (alangenfeld) [inline .buildkite/dagster-buildkite/dagster_buildkite/repository.py:23]: this worth it? feels superfluous

**Review by alangenfeld — APPROVED** [review]

File: .buildkite/dagster-buildkite/dagster_buildkite/repository.py (hunk 2 above)
Comment (jmsanders) [reply .buildkite/dagster-buildkite/dagster_buildkite/repository.py:23]: There's some ordering that's important here - PythonPackages depends on ChangedFiles. I guess I could move the `ChangedFiles.load_repository()` into PythonPackages though.

Comment (jmsanders) [reply .buildkite/dagster-buildkite/dagster_buildkite/repository.py:23]: But it does give us a nice place to group related actions for logging purposes.

Comment (alangenfeld) [reply .buildkite/dagster-buildkite/dagster_buildkite/repository.py:23]: maybe just collapse everything in to one class?

Comment (jmsanders) [reply .buildkite/dagster-buildkite/dagster_buildkite/repository.py]: I ended up making the dependency explicit within PythonPackages.

**Pull request merged**